modules/core/src/smithy4s/Endpoint.scala

```diff
@@ -39,131 +41,48 @@ package smithy4s
   * datatypes that typically fit in memory with concerns of streaming (which can
   * be encoded a great many ways, using a great many libraries)
   */
-trait Endpoint[Op[_, _, _, _, _], I, E, O, SI, SO]
-    extends Endpoint.Base[I, E, O, SI, SO] {
+// scalafmt: {maxColumn = 120}
+trait Endpoint[Op[_, _, _, _, _], I, E, O, SI, SO] {
+
+  final def mapSchema(
+      f: OperationSchema[I, E, O, SI, SO] => OperationSchema[I, E, O, SI, SO]
+  ): Endpoint[Op, I, E, O, SI, SO] = Endpoint(f(schema), wrap)
 
+  def schema: OperationSchema[I, E, O, SI, SO]
```

Comment:
this is now using composition over inheritance. Means we can have a `mapSchema` easily, rendering the builder pattern obsolete.